src/analyze/mod.rs

```diff
@@ -1417,7 +1414,7 @@ pub(crate) mod test {
     }
     fn assert_extern_decl_display(s: &str) {
         // TODO: this `auto` is such a hack
-        assert_decl_display(s, &format!("{}", s));
+        assert_decl_display(s, &s.to_string());
```

Comment:
This whole part of the code is a mess.
```suggestion
        assert_decl_display(s, s);
```